generators/server/templates/src/main/java/package/config/_LoadBalancedResourceDetails.java

```diff
@@ -0,0 +1,59 @@
+package <%=packageName%>.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.client.loadbalancer.LoadBalancerClient;
+import org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails;
+import org.springframework.stereotype.Component;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+
+@Component
+public class LoadBalancedResourceDetails extends ClientCredentialsResourceDetails {
+
```

Comment:
As discussed, it would be better to create the LoadBalancedResourceDetails class without @Component nor jhipster dependency and then create the bean in OAuth2InterceptedFeignConfiguration with
```java
@Bean
@Configurationproperties("oauthclient.security.oauth2")
LoadBalancedResourceDetails loadBalancedResourceDetails () { 
    return new LoadBalancedResourceDetails();
}

@Bean
public RequestInterceptor getOAuth2RequestInterceptor() {
    return new OAuth2FeignRequestInterceptor(new DefaultOAuth2ClientContext(), loadBalancedResourceDetails());
}
```
That way there is no risk that LoadBalancedResourceDetails is scanned for something other than the feign configuration.